zebra-chain/src/commitments/sapling/test_vectors.rs

```diff
@@ -0,0 +1,285 @@
+// Test vector data generated from
+// https://github.com/zcash-hackworks/zcash-test-vectors/blob/master/sapling_pedersen.py
+//
+// These vectors in particular correspond to the Personalization::NoteCommitment
+// enum variant from the original source.
+//
+// The Python hex-encoded outputs for these test vectors were output in
+// big-endian byte order, so to parse them, we have to reverse their order; in
+// librustzcash, they match their Display impl to match the Python hex strings
+// and that's what they compare in their unit tests, not the bytes.
```

Comment:
Ooof. Can we leave this behind and pre-reverse the test vectors before committing them to our source tree, so that we don't have to worry about inconsistent choices inside zebra-chain, and spare future users of the test vectors from the same confusion?

Reply:
I considered that, but also considered the grepability of using the same bytestring test vectors between implementations. I could also, instead, commit a comment of 'the original hex-encoded big-endian test vector from librustcash is <this>, and we reversed it and encoded it to <that>', so the grepability remains.

Reply:
Hmm, I feel slightly better about the comment but I think the cleanest thing would be to avoid it entirely or submit a fix to librustzcash -- putting wrong-endianness data in our code feels like propagating a mistake.

Reply:
Fixing upstream both includes fixing the python code in zcash-hackworks that generates the Rust-wrapped test vectors and that output that has been committed into librustzcash, so for now I figure we fix ours, include the original wrong-endianness string as a comment, and uh, suggest fixing all the upstreams.

Reply:
Also apparently this is all Bitcoin's fault:
https://twitter.com/str4d/status/1290879867018334208


![star-trek-kirk-khan-bitcoin](https://user-images.githubusercontent.com/552961/89468497-9a51f880-d745-11ea-9bb2-416c1daee0f4.gif)

Reply:
Just regenerate the test vectors with the sensible byte order; the code has an option explicitly for that.

Reply:
Ah, if the script can generate the test vectors with the correct byte order, I think we should do that and omit any wrong-order vectors (incl. in comments).

Reply:
@daira where? The hackworks repo does not have a 1:1 script to produce these vectors from these input bits (anymore?), I have to like, write code to accomplish this and I don't know where to pass this byte order arg.

Reply:
All the scripts that generate test vectors in https://github.com/zcash-hackworks/zcash-test-vectors take a `--target=` option:
* `--target=zcash` means JSON, byte-reversed
* `--target=json` means JSON, no byte reversal
* `--target=rust` means Rust, no byte reversal.

Reply:
OK resolved in https://github.com/ZcashFoundation/zebra/pull/596/commits/5143552e6832b0ec4792e99a01d5ed1fd89f0ef7